go-controller/pkg/ovn/zone_interconnect/interconnect.go

```diff
@@ -0,0 +1,524 @@
+		Name: logicalSwitchName,
+	}
+
+	logicalSwitchPort := nbdb.LogicalSwitchPort{
+		Name:      portName,
+		Type:      portType,
+		Options:   options,
+		Addresses: addresses,
+	}
+	if err := libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitch(zic.nbClient, &logicalSwitch, &logicalSwitchPort); err != nil {
+		return fmt.Errorf("failed to add logical port %s to switch %s, error: %v", portName, logicalSwitch.Name, err)
+	}
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) cleanupNodeClusterRouterPort(node *corev1.Node) error {
+	lrp := nbdb.LogicalRouterPort{
+		Name: zic.getNetworkScopedName(types.RouterToTransitSwitchPrefix + node.Name),
+	}
+	logicalRouterPort, err := libovsdbops.GetLogicalRouterPort(zic.nbClient, &lrp)
+	if err != nil {
+		// logical router port doesn't exist. So nothing to cleanup.
+		return nil
+	}
+
+	logicalRouter := nbdb.LogicalRouter{
+		Name: zic.networkClusterRouterName,
+	}
+
+	if err := libovsdbops.DeleteLogicalRouterPorts(zic.nbClient, &logicalRouter, logicalRouterPort); err != nil {
+		return fmt.Errorf("failed to delete logical router port %s from router %s for the node %s, error: %v", logicalRouterPort.Name, zic.networkClusterRouterName, node.Name, err)
+	}
+
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) cleanupNodeTransitSwitchPort(node *corev1.Node) error {
+	logicalSwitch := &nbdb.LogicalSwitch{
+		Name: zic.networkTransitSwitchName,
+	}
+	logicalSwitchPort := &nbdb.LogicalSwitchPort{
+		Name: zic.getNetworkScopedName(types.TransitSwitchToRouterPrefix + node.Name),
+	}
+
+	if err := libovsdbops.DeleteLogicalSwitchPorts(zic.nbClient, logicalSwitch, logicalSwitchPort); err != nil {
+		return fmt.Errorf("failed to delete logical switch port %s from transit switch %s for the node %s, error: %v", logicalSwitchPort.Name, zic.networkTransitSwitchName, node.Name, err)
+	}
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) setRemotePortBindingChassis(nodeName, portName, chassisId string) error {
+	remotePort := sbdb.PortBinding{
+		LogicalPort: portName,
+	}
+	chassis := sbdb.Chassis{
+		Hostname: nodeName,
+		Name:     chassisId,
+	}
+
+	if err := libovsdbops.UpdatePortBindingSetChassis(zic.sbClient, &remotePort, &chassis); err != nil {
+		return fmt.Errorf("failed to update chassis %s for remote port %s, error: %v", nodeName, portName, err)
+	}
+
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) addRemoteNodeStaticRoutes(node *corev1.Node, nodeID int, nodeTransitSwitchPortIPs []*net.IPNet) error {
+	addRoute := func(subnet *net.IPNet, tsIp string) error {
+		logicalRouterStaticRoute := nbdb.LogicalRouterStaticRoute{
+			ExternalIDs: map[string]string{
+				"ic-node": node.Name,
+			},
+			Nexthop:  tsIp,
+			IPPrefix: subnet.String(),
+		}
+		p := func(lrsr *nbdb.LogicalRouterStaticRoute) bool {
+			return lrsr.IPPrefix == subnet.String() &&
+				lrsr.Nexthop == tsIp &&
+				lrsr.ExternalIDs["ic-node"] == node.Name
+		}
+		//FIXME: All routes should be added in a single transaction instead.
```

Comment:
+1 we should do this in this go itself?

Reply:
I don't think its a major thing or anything to be concerned about.

I've already put some comments like this - // Possible optimization: Add all the routes in one transaction
so I'll remove this one.